dts: x86: remove mem.h

SavinayDharmappa opened the pull request:
patch removes the mem.h and marcos used in that file are
moved appropriate board files.

Signed-off-by: Savinay Dharmappa <savinay.dharmappa@intel.com>

galak reviewed (changes requested):
Can you explain this further in the commit message.  Why are the flash & ram sizes a function of the board and not SoC?

SavinayDharmappa committed:
dts: x86: remove mem.h

patch removes the mem.h and marcos used in that file are
moved appropriate board files. As there are boards with
different flash configuration but of same soc, flash and
ram size are moved to dts file instead of dtsi

Signed-off-by: Savinay Dharmappa <savinay.dharmappa@intel.com>

nashif commented:
please rebase
